src/Command/SynchronizeProductsCommand.php

```diff
@@ -0,0 +1,104 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\SyliusProducerPlugin\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\SyliusProducerPlugin\Producer\ProductMessageProducerInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SynchronizeProductsCommand extends BaseSynchronizeCommand
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var ProductMessageProducerInterface
+     */
+    private $productMessageProducer;
+
+    /**
+     * @var ProductRepositoryInterface
+     */
+    private $productRepository;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        ProductMessageProducerInterface $productMessageProducer,
+        ProductRepositoryInterface $productRepository
+    ) {
+        parent::__construct($entityManager);
+
+        $this->entityManager = $entityManager;
+        $this->productMessageProducer = $productMessageProducer;
+        $this->productRepository = $productRepository;
+    }
+
+    protected function configure()
+    {
+        $this->setName('sulu-sylius:synchronize:products')
+            ->setDescription('Synchronize all products to Sulu');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        parent::execute($input, $output);
+
+        $this->syncProducts($output);
+    }
+
+    private function syncProducts(OutputInterface $output): void
+    {
+        $output->writeln('<info>Sync products</info>');
+
+        $count = $this->entityManager->createQueryBuilder()
+            ->select('count(product.id)')
+            ->from($this->productRepository->getClassName(), 'product')
+            ->getQuery()
+            ->getSingleScalarResult();
+
+        $query = $this->entityManager->createQueryBuilder()
+            ->select('product')
+            ->from($this->productRepository->getClassName(), 'product')
+            ->getQuery();
+        $iterableResult = $query->iterate();
+
+        $progressBar = new ProgressBar($output, intval($count));
+        $progressBar->start();
+
+        $processedItems = 0;
+        while (($row = $iterableResult->next()) !== false) {
+            $product = $row[0];
+            if (!$product instanceof ProductInterface) {
+                continue;
+            }
+
+            $this->productMessageProducer->synchronize($product, false);
+
+            $this->entityManager->detach($product);
+            $processedItems++;
+            if ($processedItems % self::BULK_SIZE === 0) {
+                $this->entityManager->clear();
+                gc_collect_cycles();
+            }
+
+            $progressBar->advance();
+        }
+    }
```

Comment:
would add `$progressBar->finish()` to be sure that the progressbar is full at the end of the command